src/main/java/org/opensearch/knn/index/codec/KNN990Codec/KNNQuantizationStateReader.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.KNN990Codec;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.IndexFileNames;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Reads quantization states
+ */
+public class KNNQuantizationStateReader {
+
+    /**
+     * Read quantization states and return list of fieldNames and bytes
+     * @param state the read state to read from
+     */
+    public Map<String, byte[]> read(SegmentReadState state) {
+        String quantizationStateFileName = IndexFileNames.segmentFileName(state.segmentInfo.name, state.segmentSuffix, "qs");
+        Map<String, byte[]> readQuantizationStateInfos = new HashMap<>();
+
+        try (IndexInput input = state.directory.openInput(quantizationStateFileName, IOContext.READ)) {
```

Comment:
Should we open with checksum? https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/store/Directory.java#L147-L156

Reply:
I don't think that will work because I'm seeking backwards at times during the file read

Reply:
https://github.com/opensearch-project/k-NN/pull/1997

Reply:
Oh good to know then thats good